File: 1/swift/TestCalculator.swift
import Foundation

func calculate(_ x: Int, _ y: Int) -> Int {
    if x > 0 && y > 0 {
        return x + y
    } else if x > 0 && y <= 0 {
        return x - y
    } else if x <= 0 && y > 0 {
        return -x + y
    } else {
        return -x - y
    }
}

func funcTest(_ a: Int, _ b: Int, _ x: Int) -> Int {
    var result = x
    if a < 9 && b == 5 {
        result = x * a + b
    }
    if a > 2 || result > 6 {
        result += 1
    }
    return result
}

func testCalculate() {
    let testCases = [
        (5, 3, 8),
        (5, -3, 8),
        (-5, 3, 8),
        (-5, -3, 8),
        (0, 5, 5),
        (5, 0, 5),
        (0, 0, 0)
    ]
    
    print("=== Swift Calculate Tests ===")
    var allPassed = true
    
    for (x, y, expected) in testCases {
        let result = calculate(x, y)
        let status = result == expected ? "✅" : "❌"
        print("\(status) calculate(\(x), \(y)) = \(result) (expected: \(expected))")
        
        if result != expected {
            allPassed = false
        }
    }
    
    if allPassed {
        print("🎉 All calculate tests passed!")
    } else {
        print("💥 Some calculate tests failed!")
    }
}

func testFuncWhiteBox() {
    let testCases = [
        (8, 5, 1, 14),
        (1, 5, 10, 16),
        (1, 5, 0, 5),
        (5, 1, 1, 2),
        (1, 1, 10, 11),
        (1, 1, 1, 1),
        (9, 5, 2, 3),
        (2, 5, 3, 12),
        (1, 1, 6, 6),
        (1, 1, 7, 8)
    ]
    
    print("\n=== Swift Func Tests (White Box) ===")
    var allPassed = true
    
    for (a, b, x, expected) in testCases {
        let result = funcTest(a, b, x)
        let status = result == expected ? "✅" : "❌"
        print("\(status) func(\(a), \(b), \(x)) = \(result) (expected: \(expected))")
        
        if result != expected {
            allPassed = false
        }
    }
    
    if allPassed {
        print("🎉 All func tests passed!")
    } else {
        print("💥 Some func tests failed!")
    }
}

testCalculate()
testFuncWhiteBox()

File: 1/php/test_calculator.php
<?php
require_once 'calculator.php';

function test_calculate(): bool {
    $tests = [
        [5, 3, 8],
        [5, -3, 8],
        [-5, 3, 8],
        [-5, -3, 8],
        [0, 5, 5],
        [5, 0, 5],
        [0, 0, 0]
    ];
    
    echo "=== PHP Calculate Tests ===\n";
    $all_passed = true;
    
    foreach ($tests as $test) {
        list($x, $y, $expected) = $test;
        $result = calculate($x, $y);
        $status = $result === $expected ? "✅" : "❌";
        echo "$status calculate($x, $y) = $result (expected: $expected)\n";
        
        if ($result !== $expected) {
            $all_passed = false;
        }
    }
    
    if ($all_passed) {
        echo "🎉 All calculate tests passed!\n";
    } else {
        echo "💥 Some calculate tests failed!\n";
    }
    return $all_passed;
}

function test_func_white_box(): bool {
    $tests = [
        [8, 5, 1, 14], 
        [1, 5, 10, 16],
        [1, 5, 0, 5],
        [5, 1, 1, 2],
        [1, 1, 10, 11],
        [1, 1, 1, 1],
        [9, 5, 2, 3],
        [2, 5, 3, 12],
        [1, 1, 6, 6],
        [1, 1, 7, 8]
    ];
    
    echo "\n=== PHP Func Tests (White Box) ===\n";
    $all_passed = true;
    
    foreach ($tests as $test) {
        list($a, $b, $x, $expected) = $test;
        $result = func($a, $b, $x);
        $status = $result === $expected ? "✅" : "❌";
        echo "$status func($a, $b, $x) = $result (expected: $expected)\n";
        
        if ($result !== $expected) {
            $all_passed = false;
        }
    }
    
    if ($all_passed) {
        echo "🎉 All func tests passed!\n";
    } else {
        echo "💥 Some func tests failed!\n";
    }
    return $all_passed;
}

$calculate_passed = test_calculate();
$func_passed = test_func_white_box();

echo "\n=== Summary ===\n";
if ($calculate_passed && $func_passed) {
    echo "🎉 All PHP tests passed! (7 calculate + 10 func)\n";
} else {
    echo "💥 Some PHP tests failed!\n";
}
?>